omero2pandas/upload.py

```diff
@@ -66,30 +81,117 @@ def generate_omero_columns(df):
             raise NotImplementedError(f"Column type "
                                       f"{column_type} not supported")
         if col_class == omero.grid.StringColumn:
+            string_columns.append(column_name)
             max_len = df[column_name].str.len().max()
             if math.isnan(max_len):
                 max_len = 1
             col = col_class(cleaned_name, "", int(max_len), [])
-            # Coerce missing values into strings
-            df[column_name].fillna('', inplace=True)
         else:
             col = col_class(cleaned_name, "", [])
         omero_columns.append(col)
-    return omero_columns
+    return omero_columns, string_columns
 
 
-def create_table(df, table_name, parent_id, parent_type, conn, chunk_size):
+def generate_omero_columns_csv(csv_path, chunk_size=1000):
+    # Inspect a CSV file to generate OMERO.tables columns
+    LOGGER.info(f"Inspecting {csv_path}")
+    scan = pd.read_csv(csv_path, nrows=chunk_size or 1000)
+    LOGGER.debug(f"Shape is {scan.shape[0]}x{scan.shape[1]}")
+    if chunk_size is None:
+        chunk_size = optimal_chunk_size(len(scan.columns))
+    LOGGER.debug(f"Using chunk size {chunk_size}")
+    omero_columns = []
+    to_resolve = {}
+    for idx, (column_name, column_type) in enumerate(scan.dtypes.items()):
+        cleaned_name = column_name.replace('/', '\\')
+        if column_name in SPECIAL_NAMES and column_type.kind == 'i':
+            col_class = SPECIAL_NAMES[column_name]
+        elif column_type.kind in COLUMN_TYPES:
+            col_class = COLUMN_TYPES[column_type.kind]
+        else:
+            raise NotImplementedError(f"Column type "
+                                      f"{column_type} not supported")
+        if col_class == omero.grid.StringColumn:
```

Comment:
As below, import this? There are also features in `omero.columns` (which has the concrete implementations of the `omero.grid` column types that you may find useful.

Reply:
We use a lot of classes from omero.grid and omero.model, so I'd just left it as a full import.

I wasn't aware that `omero.columns` existed, it's not mentioned in the main Python docs at all. Probably worth discussing whether there's any benefit to be had from using those instead of grid, and if so it could be best as a dedicated PR.